server/statistics/hot_peer_cache.go

```diff
@@ -224,6 +180,59 @@ func (f *hotPeerCache) CheckRegionFlow(region *core.RegionInfo) (ret []*HotPeerS
 	return ret
 }
 
+// CheckPeerFlow checks the flow information of a peer.
+// Notice: CheckPeerFlow couldn't be used concurrently.
+func (f *hotPeerCache) CheckPeerFlow(peer *core.PeerInfo, region *core.RegionInfo, interval uint64) *HotPeerStat {
+	storeID := peer.GetStoreID()
+	deltaLoads := f.getFlowDeltaLoads(peer)
+	loads := make([]float64, len(deltaLoads))
+	for i := range deltaLoads {
+		loads[i] = deltaLoads[i] / float64(interval)
+	}
+	justTransferLeader := f.justTransferLeader(region)
+	// transfer read leader or remove write peer
+	isExpired := f.isPeerExpired(peer, region)
+	oldItem := f.getOldHotPeerStat(region.GetID(), storeID)
+	if isExpired && oldItem != nil && !f.isInheritExist(region.GetID()) {
```

Comment:
why need to judge whether exist in inherit items?

Reply:
removed